### ISSUE TITLE
Epsilon-Greedy Exploration in Hybrid DDPG

### ISSUE THREAD

**Issue by MarkHolmstrom** [opendilab/DI-engine#847]
The wrappers used for handling a hybrid action space in the DDPG collect model https://github.com/opendilab/DI-engine/blob/548406fa10d6082431480f53cfba2cb20da064fc/ding/policy/ddpg.py#L399 and learn/target models https://github.com/opendilab/DI-engine/blob/548406fa10d6082431480f53cfba2cb20da064fc/ding/policy/ddpg.py#L190 seem to be contradictory. Epsilon is used to select a random discrete action in the collect wrapper by modifying `action_type` in the `output` dictionary https://github.com/opendilab/DI-engine/blob/548406fa10d6082431480f53cfba2cb20da064fc/ding/model/wrapper/model_wrappers.py#L737 which is used for action selection in the agent's interaction with the environment, however in policy learning `action_type` is overriden by taking the argmax of logits. https://github.com/opendilab/DI-engine/blob/548406fa10d6082431480f53cfba2cb20da064fc/ding/model/wrapper/model_wrappers.py#L517
The model is likely to get a different action from the one taken in the environment as a result. 

In [Deep Reinforcement Learning in Parameterized Action Space](https://arxiv.org/pdf/1511.04143#subsection.4.1) the authors suggest Epsilon-Greedy sampling the logits from a uniform distribution, which I think would make sense to have in the collect wrapper instead of the current implementation.

- [x] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [x] RL algorithm bug
    + [ ] system worker bug
    + [ ] system utils bug
    + [ ] code design/refactor
    + [ ] documentation request
    + [ ] new feature request
- [x] I have visited the [readme](https://github.com/opendilab/DI-engine/blob/github-dev/README.md) and [doc](https://opendilab.github.io/DI-engine/)
- [x] I have searched through the [issue tracker](https://github.com/opendilab/DI-engine/issues) and [pr tracker](https://github.com/opendilab/DI-engine/pulls)
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import ding, torch, sys
  print(ding.__version__, torch.__version__, sys.version, sys.platform)
  ```

**Comment by puyuan1996** [comment]
---

Hello, thank you for your attention to the **HybridDDPG** implementation details and for your feedback!

Regarding the issue you mentioned about using different wrappers in the `collect` and `learn` phases, our implementation logic is as follows:

### 1. **Collect Phase**  
In `self._collect_model`, we use the `eps_greedy_multinomial` variant. Unlike the traditional Epsilon-Greedy approach, this variant works as follows:  

- With a probability of `eps`, it selects an action uniformly at random.  
- With a probability of `1-eps`, it samples an action from the current policy's distribution using `torch.multinomial` (i.e., sampling actions based on the probabilities represented by the logits).  

The reason for choosing this implementation is that, during experiments, we observed that this approach enables more efficient exploration of the environment compared to the traditional Epsilon-Greedy approach (which directly selects the maximum logit or samples randomly). This enhanced exploration improves the efficiency of the policy's learning process.

### 2. **Learn Phase**  
In `self._learn_model`, we use `hybrid_argmax_sample`. This is because the training of the Q-network requires the calculation of the TD-target, which depends on `argmax Q(s', a')` (i.e., the action with the maximum Q-value in the target state). Therefore, in `self._learn_model`, `hybrid_argmax_sample` is used to return the discrete action that maximizes the Q-value in the current state. For reference, you can check the implementation in [model_wrappers.py#L520](https://github.com/opendilab/DI-engine/blob/548406fa10d6082431480f53cfba2cb20da064fc/ding/model/wrapper/model_wrappers.py#L520).  

---

If anything remains unclear or if you have further questions, feel free to reach out! We truly appreciate your attention and support! 😊

**Comment by MarkHolmstrom** [comment]
My confusion was over the use of the same wrapper for both the target and learn models, with how the argmax would be used for calculating `Q(s,a)`, but I see now that the wrapper has different functionality for that output https://github.com/opendilab/DI-engine/blob/56158164f97fe2801987167c7d2f16b763332ceb/ding/model/wrapper/model_wrappers.py#L508
